homeassistant/components/blink/__init__.py

```diff
@@ -1,153 +1,161 @@
-
-TYPE_CAMERA_ARMED = "motion_enabled"
-TYPE_MOTION_DETECTED = "motion_detected"
-TYPE_TEMPERATURE = "temperature"
-TYPE_BATTERY = "battery"
-TYPE_WIFI_STRENGTH = "wifi_strength"
-
-SERVICE_REFRESH = "blink_update"
-SERVICE_TRIGGER = "trigger_camera"
-SERVICE_SAVE_VIDEO = "save_video"
-
-BINARY_SENSORS = {
-    TYPE_CAMERA_ARMED: ["Camera Armed", "mdi:verified"],
-    TYPE_MOTION_DETECTED: ["Motion Detected", "mdi:run-fast"],
-}
-
-SENSORS = {
-    TYPE_TEMPERATURE: ["Temperature", TEMP_FAHRENHEIT, "mdi:thermometer"],
-    TYPE_BATTERY: ["Battery", "", "mdi:battery-80"],
-    TYPE_WIFI_STRENGTH: ["Wifi Signal", "dBm", "mdi:wifi-strength-2"],
-}
-
-BINARY_SENSOR_SCHEMA = vol.Schema(
-    {
-        vol.Optional(CONF_MONITORED_CONDITIONS, default=list(BINARY_SENSORS)): vol.All(
-            cv.ensure_list, [vol.In(BINARY_SENSORS)]
-        )
-    }
+from homeassistant.helpers import config_validation as cv
+
+from .const import (
+    BLINK_CONFIG,
+    DEFAULT_OFFSET,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    PLATFORMS,
+    SERVICE_REFRESH,
+    SERVICE_SAVE_VIDEO,
+    SERVICE_SEND_PIN,
+    SERVICE_TRIGGER,
 )
 
-SENSOR_SCHEMA = vol.Schema(
-    {
-        vol.Optional(CONF_MONITORED_CONDITIONS, default=list(SENSORS)): vol.All(
-            cv.ensure_list, [vol.In(SENSORS)]
-        )
-    }
-)
+_LOGGER = logging.getLogger(__name__)
 
 SERVICE_TRIGGER_SCHEMA = vol.Schema({vol.Required(CONF_NAME): cv.string})
-
 SERVICE_SAVE_VIDEO_SCHEMA = vol.Schema(
     {vol.Required(CONF_NAME): cv.string, vol.Required(CONF_FILENAME): cv.string}
 )
+SERVICE_SEND_PIN_SCHEMA = vol.Schema({vol.Optional(CONF_PIN): cv.string})
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
             {
                 vol.Required(CONF_USERNAME): cv.string,
                 vol.Required(CONF_PASSWORD): cv.string,
-                vol.Optional(
-                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
-                ): cv.time_period,
-                vol.Optional(CONF_BINARY_SENSORS, default={}): BINARY_SENSOR_SCHEMA,
-                vol.Optional(CONF_SENSORS, default={}): SENSOR_SCHEMA,
-                vol.Optional(CONF_OFFSET, default=1): int,
-                vol.Optional(CONF_MODE, default=""): cv.string,
+                vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): int,
             }
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-def setup(hass, config):
-    """Set up Blink System."""
+async def async_setup(hass, config):
+    """Set up a config entry."""
+    if DOMAIN not in config:
+        return True
 
-    conf = config[BLINK_DATA]
+    conf = config.get(DOMAIN, {})
     username = conf[CONF_USERNAME]
     password = conf[CONF_PASSWORD]
     scan_interval = conf[CONF_SCAN_INTERVAL]
-    is_legacy = bool(conf[CONF_MODE] == "legacy")
-    motion_interval = conf[CONF_OFFSET]
-    hass.data[BLINK_DATA] = blinkpy.Blink(
+    hass.data[DOMAIN] = blinkpy.Blink(
```

Comment:
We shouldn't do this. When we use config entries, everything needs to go via a config entry. There should be no initialization happening anymore in async_setup.